lib/matplotlib/axis.py

```diff
@@ -719,24 +733,17 @@ def get_children(self):
         children.extend(minorticks)
         return children
 
-    def cla(self):
+    def cla(self, shared=None):
         'clear the current axis'
-        self.set_major_locator(mticker.AutoLocator())
-        self.set_major_formatter(mticker.ScalarFormatter())
-        self.set_minor_locator(mticker.NullLocator())
-        self.set_minor_formatter(mticker.NullFormatter())
 
-        self.set_label_text('')
-        self._set_artist_props(self.label)
+        self.label.set_text('')  # self.set_label_text would change isDefault_
+        self._set_artist_props(self.label)  # sets figure; needed here?
```

Comment:
Actually it is already in _get_label, so I don't think we need it in cla or __init__.